lib/optimizely.rb

```diff
@@ -1098,12 +1098,13 @@ def send_impression(config, experiment, variation_key, flag_key, rule_key, enabl
       experiment_id = experiment['id']
       experiment_key = experiment['key']
 
-      if experiment_id != ''
-        variation_id = config.get_variation_id_from_key_by_experiment_id(experiment_id, variation_key)
-      else
-        varaition = get_flag_variation_by_key(flag_key, variation_key)
-        variation_id = varaition ? varaition['id'] : ''
-      end
+      variation = get_flag_variation(flag_key, variation_key, 'key')
+
+      variation_id = if !variation
+                       experiment_id != '' ? config.get_variation_id_from_key_by_experiment_id(experiment_id, variation_key) : ''
+                     else
+                       variation ? variation['id'] : ''
+                     end
```

Comment:
```suggestion
      variation_id = variation ? variation['id'] : ''
```
If it's null from get_flag_variagtion(), then get_variation_id_from_key_by_experiment_id() will also return null.

Reply:
This is independent of `flag_key`, for `a/b` experiments this will return non-null and `get_flag_variation` will return null.